tests/test_utils.py

```diff
@@ -66,6 +66,40 @@ def test_file(self, tmpdir, mode, warnings_num, recwarn):
             assert super_len(fd) == 4
         assert len(recwarn) == warnings_num
 
+    def test_super_len_with__len__(self):
+        foo = [1,2,3,4]
+        len_foo = super_len(foo)
+        assert len_foo == 4
+
+    def test_super_len_with_no__len__(self):
+        class LenFile(object):
+            def __init__(self):
+                self.len = 5
+
+        foo = LenFile()
+        assert super_len(foo) == 5
+
+    def test_super_len_with_tell(self):
+        foo = StringIO.StringIO('12345')
+        assert super_len(foo) == 5
+        foo.read(2)
+        assert super_len(foo) == 3
+
+    def test_super_len_with_fileno(self):
+        f = open(__file__, 'rb')
```

Comment:
Definitely, I somehow got it in my head that context managers didn't work in 2.6 because you can't use them with assertRaises() in 2.6. Updated.